hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -233,26 +232,21 @@ define([
 
                     ContractService.updateHeaderInfo();
                     updateContractView(results);
+                    $scope.updateContractList(results.details.period_end_date);
 
                     if (results.revisionCreated) {
 
-                        var dateEffectiveRevisionCreated = new Date(results.revisionCreated.effective_date).setHours(0, 0, 0, 0),
-                            dateEffectiveRevisionCurrent = new Date($scope.revisionCurrent.effective_date).setHours(0, 0, 0, 0),
-                            dateToday = new Date().setHours(0, 0, 0, 0),
+                        var dateEffectiveRevisionCreated = moment(new Date(results.revisionCreated.effective_date)),
+                            dateEffectiveRevisionCurrent = moment(new Date($scope.revisionCurrent.effective_date)),
+                            dateToday = moment(),
                             revisionData = {
                                 revisionEntityIdObj: results.revisionCreated,
                                 details: results.details,
                                 hour: results.hour,
                                 pay: results.pay
                             },
-                            isCurrentRevision = (dateEffectiveRevisionCreated <= dateToday &&
-                                dateEffectiveRevisionCreated >= dateEffectiveRevisionCurrent) ||
-                                (dateEffectiveRevisionCurrent > dateToday &&
-                                dateEffectiveRevisionCreated <= dateEffectiveRevisionCurrent);
+                            isCurrentRevision = dateEffectiveRevisionCurrent.diff(dateToday, "day") <= 0 || dateEffectiveRevisionCurrent.diff(dateEffectiveRevisionCreated, "day") <= 0;
 
-                        if (isCurrentRevision) {
-                            updateContractList(results.details.period_end_date);
```

Comment:
It's safe to remove. What is means is that with or without a revision the contact list will update, this is because execution of the  `updateContractList` has been moved to the top.